presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2751,14 +2754,46 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
             predicate = createPredicate(partitionColumns, partitions);
         }
 
+        // Expose ordering property of the table.
+        ImmutableList.Builder<LocalProperty<ColumnHandle>> localProperties = ImmutableList.builder();
+        Optional<Set<ColumnHandle>> streamPartitionColumns = Optional.empty();
+        if (table.getStorage().getBucketProperty().isPresent() && !table.getStorage().getBucketProperty().get().getSortedBy().isEmpty()) {
+            ImmutableSet.Builder<ColumnHandle> streamPartitionColumnsBuilder = ImmutableSet.builder();
+
+            // streamPartitioningColumns is how we partition the data across splits.
+            // localProperty is how we partition the data within a split.
+            // 1. add partition columns to streamPartitionColumns
+            partitionColumns.forEach(streamPartitionColumnsBuilder::add);
+
+            // 2. add sorted columns to streamPartitionColumns and localProperties
+            HiveBucketProperty bucketProperty = table.getStorage().getBucketProperty().get();
+            Map<String, ColumnHandle> columnHandles = hiveColumnHandles(table).stream()
+                    .collect(toImmutableMap(HiveColumnHandle::getName, identity()));
+            bucketProperty.getSortedBy().forEach(sortingColumn -> {
+                ColumnHandle columnHandle = columnHandles.get(sortingColumn.getColumnName());
+                localProperties.add(new SortingProperty<>(columnHandle, sortingColumn.getOrder().getSortOrder()));
+                streamPartitionColumnsBuilder.add(columnHandle);
+            });
+
+            // We currently only set streamPartitionColumns when it enables streaming aggregation and also it's eligible to enable streaming aggregation
+            // 1. When the bucket columns are the same as the prefix of the sort columns
+            // 2. When all rows of the same value group are guaranteed to be in the same split. We disable splitting a file when isStreamingAggregationEnabled is true to make sure the property is guaranteed.
+            List<String> sortColumns = bucketProperty.getSortedBy().stream().map(SortingColumn::getColumnName).collect(toImmutableList());
+            if (bucketProperty.getBucketedBy().size() <= sortColumns.size()
+                    && bucketProperty.getBucketedBy().containsAll(sortColumns.subList(0, bucketProperty.getBucketedBy().size()))
```

Comment:
Does the prefix match needs to be ordered ? I assume equality condition and hence order does not matter.

Reply:
Right, the order doesn't not matter here. StreamPartitionColumns is similar to the bucket column concept by it is for splits instead of files.